Sources/Basic/FileSystem.swift

```diff
@@ -664,7 +664,7 @@ public class InMemoryFileSystem: FileSystem {
         // Ignore root and get the parent node's content if its a directory.
         guard !path.isRoot,
               let parent = try? getNode(path.parentDirectory),
-              case .directory(let contents)? = parent?.contents else {
+              case .directory(let contents) = parent.contents else {
```

Comment:
You still need `?` in `parent?.contents` because it's optional.

Reply:
```
S:\swift-package-manager\Sources\Basic\FileSystem.swift:667:53: error: cannot use optional chaining on non-optional value of type 'InMemoryFileSystem.Node'
              case .directory(let contents) = parent?.contents else {
                                              ~~~~~~^
```

Reply:
Looks like Swift compiler on Windows is wrong here? `getNode` returns an optional and you get double optional because of `try?` but `guard let` removes one Optional so you should be left with one Optional in end. We can probably tweak the logic here to avoid this issue but we should file a bug on the compiler.

Reply:
IIRC 5.0 or 5.1 introduced `try` coalescing, such that the result of a `try?` flattens multiple levels of optional into just one. I don’t recall how accurate this is, but I remember there was some discussion around it.

Reply:
@millenomi - spot on!  I missed the language version flag in the Package.swift.